---
layout: default
title: Home
nav_order: 1
description: "SCoPE2: Single-cell protein analysis by mass spectrometry; second generation SCoPE-MS"
permalink: /
---
{% include social-media-links.html %}

# **SCoPE2**:

{: .fs-9 }

<u>S</u>ingle <u>C</u>ell <u>P</u>rot<u>E</u>omics by Mass Spectrometry (Version 2)
{: .fs-6 .fw-300}

### Quantifying proteins in single cells at high-throughput with mass spectrometry

[SCoPE2 Preprint](https://doi.org/10.1101/665307){: .btn .fs-5 .mb-4 .mb-md-0 .mr-2 }
[SCoPE2 Paper](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02267-5){: .btn .fs-5 .mb-4 .mb-md-0 .mr-2 }
[SCoPE2 code on GitHub](https://github.com/SlavovLab/SCoPE2/tree/master/code){: .btn .fs-5 .mb-4 .mb-md-0 }

* [Highlights]({{site.baseurl}}#highlights)
* [Reviews]({{site.baseurl}}#reviews)
* [Perspectives]({{site.baseurl}}#perspectives)


------------

The fate and physiology of individual cells are controlled by proteins. Yet, our ability to quantitatively analyze proteins in single cells has remained limited. To overcome this barrier, we developed SCoPE2. It substantially increases quantitative accuracy and throughput while lowering cost and hands-on time by introducing automated and miniaturized sample preparation. These advances enabled us to analyze the emergence of cellular heterogeneity as homogeneous monocytes differentiated into macrophage-like cells in the absence of polarizing cytokines. SCoPE2 quantified over 3,042 proteins in 1,490 single monocytes and macrophages in ten days of instrument time, and the quantified proteins allowed us to discern single cells by cell type. Furthermore, the data uncovered a continuous gradient of proteome states for the macrophage-like cells, suggesting that macrophage heterogeneity may emerge even in the absence of polarizing cytokines. Parallel measurements of transcripts by 10x Genomics scRNA-seq suggest that our measurements sampled 20-fold more protein copies than RNA copies per gene, and thus SCoPE2 supports quantification with improved count statistics. Joint analysis of the data illustrates how variability across single cells can reveal transcriptional and post-transcriptional gene regulation. Our methodology lays the foundation for automated and quantitative single-cell analysis of proteins by mass-spectrometry.

------------

![]({{site.baseurl}}/assets/images/SCOPE2-ac.png){: width="100%" .center-image}

------------

## Highlights
* [Highlight by Nature Methods](https://www.nature.com/articles/s41592-021-01243-y)
* [Technology feature at Nature Methods](https://www.nature.com/articles/s41592-019-0540-6)
* [Single-cell proteomics for the Human Cell Atlas](https://news.northeastern.edu/2019/07/08/northeastern-university-proteomics-researcher-receives-grant-from-chan-zuckerberg-initiative-to-help-map-all-cells-in-the-human-body-to-better-understand-cancer-diabetes-and-other-diseases/)
* [News in Proteomics Research](http://proteomicsnews.blogspot.com/2019/12/over-1000-single-cell-proteomes-2700.html)
* [The single cell proteomics revolution](https://www.bioanalysis-zone.com/2020/02/11/single-cell-proteomics-revolution_bo/)

## Reviews
* [Single-cell protein analysis by mass spectrometry](https://doi.org/10.1016/j.cbpa.2020.04.018), *Current Opinion in Chemical Biology*
* [Single cell protein analysis for systems biology](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6204083/),	*Essays in biochemistry*

## Perspectives
* [Transformative Opportunities for Single-Cell Proteomics](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6089608/), *Journal of Proteome Research*
* [Unpicking the proteome in single cells](https://science.sciencemag.org/content/367/6477/512), *Science*



## About the project

SCoPE2 is a project developed in the [Slavov Laboratory](http://slavovlab.net) and [SCP Center](http://center.single-cell.net) at [Northeastern University](https://www.northeastern.edu/), and was authored by [Harrison Specht](http://harrisonspecht.com), [Ed Emmott](http://edemmott.co.uk/), [Aleksandra Petelski](http://slavovlab.net/people.htm), [R. Gray Huffman](http://slavovlab.net/people.htm), [David H Perlman](http://slavovlab.net/people.htm), Marco Serra, [Peter Kharchenko](http://pklab.med.harvard.edu/), [Toni Koller](http://slavovlab.net/people.htm), and [Nikolai Slavov](https://coe.northeastern.edu/people/slavov-nikolai/).   


Contact the authors by email: [nslavov\{at\}northeastern.edu](mailto:nslavov@northeastern.edu).

This project was supported by funding from the [NIH Director's Award](https://projectreporter.nih.gov/project_info_description.cfm?aid=9167004&icde=31336575) and by an [Allen Distinguished Investigator Award](https://alleninstitute.org/what-we-do/frontiers-group/distinguished-investigators/projects/tracking-proteome-dynamics-single-cells) from the Paul G. Allen Frontiers Group.
